src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -177,5 +209,338 @@ private void ThrowIfDisposed()
+
+                subscriptionRuntime = await _managementClient.GetSubscriptionRuntimeInfoAsync(topicPath, subscriptionPath);
+                activeMessageCount = subscriptionRuntime.MessageCountDetails.ActiveMessageCount;
+                deadLetterCount = subscriptionRuntime.MessageCountDetails.DeadLetterMessageCount;
+
+                // If partitioning is turned on, then Service Bus automatically partitions queues into 16 partitions
+                // See more information here: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-partitioning#standard
+                topic = await _managementClient.GetTopicAsync(topicPath);
+                if (topic.EnablePartitioning)
+                {
+                    partitionCount = 16;
+                }
+            }
+            catch (MessagingEntityNotFoundException)
+            {
+                _logger.LogWarning($"ServiceBus subscription with path '{_entityPath}' was not found.");
+            }
+            catch (UnauthorizedAccessException) // When Manage claim is not used on Service Bus connection string
+            {
+                if (TimeToLogWarning())
+                {
+                    _logger.LogWarning($"Connection string does not have Manage claim for topic '{_entityPath}'. Failed to get subscription description to derive topic queue length metrics. " +
+                        $"Falling back to using first message enqueued time.");
+                }
+
+                isUnauthorized = true;
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning($"Error querying for Service Bus topic scale status: {e.Message}");
+            }
+
+            return CompileTriggerMetrics(message, activeMessageCount, deadLetterCount, partitionCount, isUnauthorized, _isListeningOnDeadLetterQueue);
+        }
+
+        public static ServiceBusTriggerMetrics CompileTriggerMetrics(Message message, long activeMessageCount, long deadLetterCount, int partitionCount,
+            bool isUnauthorized, bool isListeningOnDeadLetterQueue)
+        {
+            long totalNewEventCount = 0;
+            TimeSpan eventQueueTime = TimeSpan.Zero;
+
+            if (message != null)
+            {
+                eventQueueTime = DateTime.UtcNow.Subtract(message.SystemProperties.EnqueuedTimeUtc);
+                totalNewEventCount = 1;
+            }
+
+            if (!isUnauthorized && ((!isListeningOnDeadLetterQueue && activeMessageCount > 0) || (isListeningOnDeadLetterQueue && deadLetterCount > 0)))
+            {
+                totalNewEventCount = isListeningOnDeadLetterQueue ? deadLetterCount : activeMessageCount;
+            }
+
+            return new ServiceBusTriggerMetrics
+            {
+                Timestamp = DateTime.UtcNow,
+                TotalNewEventCount = totalNewEventCount,
+                PartitionCount = partitionCount,
```

Comment:
Why do we need `int` if it's always `16` or `0`? Maybe bool property?

Reply:
So there is a section of the scaling logic (GetScaleStatusCore) that needs to use the actual number to make a decision. I can change the scaling logic to make an assumption about the number of partitions if that's better - I guess I just thought including the number in the metrics was cleaner and matches the other triggers